jira_metrics/epic_tracking.py

```diff
@@ -42,142 +61,21 @@
-
-JQL_EPICS = os.environ.get("JIRA_JQL_EPICS", default_jql)
+# All epic selection is command line based - no environment variables needed
 
 # We'll validate these in main() using a proper validation function
 
-# Jira REST API v3 - Updated endpoint as per migration guide
-if SITE:
-    API_SEARCH = f"{SITE}/rest/api/3/search/jql"
-else:
-    API_SEARCH = None
 
 # Buckets (lowercase status names compared against these sets)
-DONE_STATUSES = {"done", "released"}
-INPROG_STATUSES = {"in progress", "in develop"}
-
-
-# ---------- Helpers ----------
-def _req_get(url: str, params: Dict[str, Any]) -> Dict[str, Any]:
-    """GET with basic auth + simple retry on 429/5xx - following GitHub API patterns."""
-    verbose_print(f"Making request to: {url}")
-    verbose_print(f"Request params: {params}")
-
-    headers = {
-        "Accept": "application/json",
-        "Content-Type": "application/json"
-    }
-
-    for attempt in range(5):
-        try:
-            r = requests.get(
-                url,
-                params=params,
-                auth=(EMAIL, API_TOKEN),
-                headers=headers,
-                timeout=30
-            )
-            verbose_print(f"Response status: {r.status_code}")
-
-            if r.status_code in (429, 500, 502, 503, 504):
-                wait = min(2**attempt, 10)
-                verbose_print(f"Rate limited or server error, waiting {wait}s...")
-                time.sleep(wait)
-                continue
-
-            if r.status_code != 200:
-                print(f"ERROR: Request failed with status {r.status_code}")
-                print(f"URL: {r.url}")
-                print(f"Response: {r.text[:500]}")  # Limit response text
-
-            r.raise_for_status()
-            return r.json()
-
-        except requests.exceptions.RequestException as e:
-            if attempt == 4:  # Last attempt
-                raise
-            wait = min(2**attempt, 10)
-            verbose_print(f"Request exception: {e}. Retrying in {wait}s...")
-            time.sleep(wait)
-
-    # This shouldn't be reached, but just in case
-    r.raise_for_status()
-    return r.json()
-
-
-def search_jql(jql: str, fields: List[str], max_per_page: int = 100) -> List[Dict[str, Any]]:
-    """Paginate through search results using JIRA REST API v3."""
-    if not API_SEARCH:
-        raise ValueError("API_SEARCH endpoint not configured - check JIRA_SITE environment variable")
-
-    verbose_print(f"Executing JQL: {jql}")
-    verbose_print(f"Requested fields: {fields}")
-
-    issues: List[Dict[str, Any]] = []
-    start_at = 0
-
-    while True:
-        params = {
-            "jql": jql,
-            "startAt": start_at,
-            "maxResults": max_per_page,
-            "fields": ",".join(fields),
-        }
-
-        page = _req_get(API_SEARCH, params)
-
-        if not page:
-            verbose_print("No response data received")
-            break
-
-        page_issues = page.get("issues", [])
```

Comment:
this all wasn't working great and we are now instead utilizing jira_utils.py file for this common work